### PR TITLE
need to populate more default values for 2017-07-01

### LINKED ISSUE #936: Close the gap of acs-engine api and ACS public document
**Is this a request for help?**:
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE

I am noticing the api model on acs-engine is not aligned with what published [here](https://docs.microsoft.com/en-us/rest/api/compute/containerservices). For example, acs-engine requires user pass in a valid MasterProfile.Count field, but the document clearly states that is not required, and be default to 1. 

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
All

**What happened**:
Once we rollout acs-engine with acs rp in the next one or two months, broader people will notice the breaking change even with the same api-version, such as 2016-09-30. The behavior will be different.

**What you expected to happen**:
User shouldn't notice we swapped the backend service. As we still conform the same api specs.

**How to reproduce it** (as minimally and precisely as possible):
Check 2016-09-30's validate function on `MasterProfile.Count`, apparently 0 is not allowed. The document allows 0, or empty.

### PR DESCRIPTION
**Pull request by rjtsdl**
**What this PR does / why we need it**:
Need to populate more default values for 2017-07-01

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #936 

**Special notes for your reviewer**:
Ready to review

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/951)

### CODE REVIEW

File: pkg/api/v20170701/types.go [hunk 1]
```diff
@@ -139,9 +139,18 @@ func (m *MasterProfile) UnmarshalJSON(b []byte) error {
...
+		m.FirstConsecutiveStaticIP = "10.240.255.5"
+	}
+
+	if m.StorageProfile == "" {
+		// if StorageProfile is missing, set to default StorageAccount
+		m.StorageProfile = StorageAccount
+	}
+
+	// OSDiskSizeGB
+	// The current behavior is, when OSDiskSizeGB is 0, we just don't pass it in the arm template
```
Comment (rjtsdl) [inline pkg/api/v20170701/types.go]: @anhowe @amanohar ,
Wondering what the default value would be used for OSDiskSizeGB? OSDiskSizeGB is different with other fields, such as "StorageProfile". I can deduce StorageProfile from either template, or SetDefaultProperties function. However, OSDiskSizeGB will be strip out in the arm template if it is 0.

Options:
1. We don't have a default value defined in the api spec. (I don't like this. User would be confused)
2. We pick a value as default. (What the value would be 200GB?)

Comment (JackQuincy) [inline pkg/api/v20170701/types.go]: OSDIskSizeGB is an override value. vm sizes have default OS disk sizes. We should leave it at 0 if it isn't set and the user should get the default for the vm size.

Comment (rjtsdl) [inline pkg/api/v20170701/types.go]: I see. I think i can change the comment to be what you have here.

**Comment by rjtsdl** [comment]
@acs-bot test this please

**Comment by rjtsdl** [comment]
@acs-bot test this please

**Pull request merged**